main.rb

```diff
@@ -1,9 +1,22 @@
-require_relative 'person'
-require_relative 'capitalize_decorator'
-require_relative 'trimmer_decorator'
+require_relative 'app'
 
-person = Person.new(22, name: 'maximilianus')
-capitalized_person = CapitalizeDecorator.new(person)
-puts capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-puts capitalized_trimmed_person.correct_name
+def main
+  app = App.new
+  app.run
+end
+
```

Comment:
- [ ] As suggested by all the previous reviewers Please refactor the code to separate user interaction from core application logic by moving the `print_options_menu` method from app.rb to main.rb and ensuring that app.rb only contains methods that implement the application's functionality. This will improve the code's maintainability and make it easier to understand. example snippet:

```rb
# main.rb
def main
  app = App.new()

  response = nil

  # [OPTIONAL] you can remind the student about the ruby lessons about basic sintax like loops: https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/lessons/basic_syntax.md  
  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

# 'main' method should be invoked at the end of the main.rb file, hint in this lesson https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point
main()


```

Reply:
@shella12  please can we meet in zoom to discuss  this?

Reply:
To be honest i still do not understand what needs doing because the app.rb  has both pubic and private methods and the main.rb file is the entry point of the app which when invoked, present the user with these options, I must not write the code the way you want it except there is only one wey of solving a programming program.

Reply:
If you look closely at my project, you will see that i have sperated responsibilty between several files. How should refactor code be a project requirement which does not even stated.